torch/_dynamo/backends/distributed.py

```diff
@@ -444,17 +436,30 @@ def run_node(self, n: Node) -> Any:
 
                     # Finally, we have to produce inputs for use compiling the next submodule,
                     # and these need to be FakeTensors, so we execute the module under fake_mode
-                    with fake_mode:
-                        return curr_submod(*new_args, **kwargs)
+                    return curr_submod(*args, **kwargs)
                 else:
                     # placeholder or output nodes don't need to get compiled, just executed
-                    return getattr(self, n.op)(n.target, new_args, kwargs)
+                    return getattr(self, n.op)(n.target, args, kwargs)
+
+        class DDPOptimizedModule(torch.nn.Module):
+            def __init__(self, split_gm):
+                super().__init__()
+                self.split_gm = split_gm
+                self.compiled = False
```

Comment:
Curious will using nn.Module to represent the delayed compilation logic affect the fully-qualified name of the original module? (mainly for my own learning)

Reply:
Yea we cannot do this, I think subclassing from OptimizedModule instead might be a solution.